### DISCUSSION TITLE
Authentication using AWS Secrets Manager on Windows other than env variable

### DISCUSSION THREAD

**Discussion by anova-git** [saleweaver/python-amazon-sp-api#1560] [General]
Is it possible to provide SP_API_AWS_SECRET_ID variable value from the credentials.yml file?
Getting the credentials from the AWS Secrets Manager works fine when the SP_API_AWS_SECRET_ID variable is an environment variable in Windows. My issue is that changing environmental variables requires a reboot, and their security is not my first choice.

What works for me:
Secrets Manager: SP_API_AWS_SECRET_ID as environment variable

Providing credentials in the code, from the credentials.yml file, or the environment variables

        refresh_token='your_refresh_token',
        lwa_app_id='your_lwa_app_id',
        lwa_client_secret='your_lwa_client_secret'

What does not work for me:
Secrets Manager: SP_API_AWS_SECRET_ID variable value from the credentials.yml file

**Comment by saleweaver** [comment]
Hi, 

you could potentially set the env variable from code, **before** importing the library. 

    os.environ[...]='your secret'

Providing the value via the config file could work too, do you want to submit a PR?

**Comment by anova-git** [comment]
Rewriting my previous post: I re-check how to get an environment variable to work in Windows without reboot, and following Michel's suggestion, 

> you could potentially set the env variable from code, before importing the library.

I tested it, and it worked. For now, the solution is good enough.

Thanks,

Marco R.
